### PR TITLE
Update delete

### PR DESCRIPTION
**Pull request by andersonlebon**
- [ ] Remove all hardcoded items from the tasks array.
- [ ] Create a new JavaScript file for the new functionality.
- [ ] Implement a function for adding a new task (add a new element to the array).
- [ ] Implement a function for editing task descriptions.
- [ ] Implement a function for deleting a task (remove an element from the array).
- [ ] Implement a function for the "Clear all completed" button (use filter() method).

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @andersonlebon, 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._
- [ ] Functionalities don't work due to an error on the console.

Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
 const addIcon = document.querySelector('.add-icon');
 const addTaskInput = document.querySelector('.add-taskInput input');
 
+const allData = Data.getDataAll();
+let index = 0;
+if (allData !== null) {
+  allData.forEach((adata) => {
+    adata.index = index;
+    index += 1;
+  });
```
Comment (mikenath223) [inline src/index.js:17]: - [ ]  There is an error in the console, preventing the project from working: `TypeError: Cannot read property 'forEach' of null`. Perhaps optional chaining would help: `allData?.forEach`, to handle a situation where no item exists in the localStorage on initial page load.

Comment (andersonlebon) [reply src/index.js:17]: Hello, @mikenath223 Yes I realized it before. And corrected it maybe you did not check my last commit.

**Review by abruzy — APPROVED** [review]
```markdown
Hi @andersonlebon 👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚢  
Congratulations! 🎉 

[OPTIONAL]
- it would be nice to clear the input field after adding items to the list.
- you can also consider implementing functionality like keypress, when I press on the enter key, I should be able to submit my item but in your case, it doesn't work, I have to use my mouse to click the plus icon, so it would be nice to have this functionality in your app.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**